src/configuration/CertificateDetailsModal.tsx

```diff
@@ -0,0 +1,358 @@
+    let result = '';
+    if (typeof ext[key] === 'object') {
+      result += JSON.stringify(ext[key]);
+    } else {
+      result += ext[key].toString();
+    }
+
+    return result;
+  };
+
+  const getExtensionsList = (crt: pki.Certificate, order: number) => {
+    return [
+      crt.extensions.map((ext, index) => (
+        <>
+          <Split key={'extension' + index + 'order' + order}>
+            <SplitItem>{index + 1} -</SplitItem>
+            <SplitItem isFilled>
+              <b>{ext.name}</b>
+            </SplitItem>
+          </Split>
+          {[
+            Object.keys(ext).map(
+              (key) =>
+                key !== 'name' &&
+                key !== 'value' && (
+                  <Split hasGutter key={ext.name + key}>
+                    <SplitItem></SplitItem>
+                    <SplitItem>
+                      <b>{key}: </b>
+                    </SplitItem>
+                    <SplitItem isFilled>
+                      <code>{getExtPropContent(ext, key)}</code>
+                    </SplitItem>
+                  </Split>
+                ),
+            ),
+          ]}
+        </>
+      )),
+    ];
+  };
+
+  const formatArray = (obj: Array<any>, indent = ''): string => {
+    let result = '';
+    const width = 6;
+    let wcnt = 0;
+    result += indent;
+    for (let i = 0; i < obj.length; i++) {
+      result += obj[i] + ',';
+      wcnt++;
+      if (wcnt === width) {
+        wcnt = 0;
+        result += '\n' + indent;
+      }
+    }
+    return result + '\n';
+  };
+
+  const formatObject = (obj: any, indent = ''): string => {
+    let result = '';
+    if (obj instanceof Array) {
+      const output = formatArray(obj, indent);
+      result += output;
+    } else if (obj instanceof Object) {
+      for (const [key, value] of Object.entries(obj)) {
+        if (value instanceof Function) {
+          continue;
+        }
+        result += indent + key + ':';
+        if (value instanceof Object) {
+          result += '\n' + formatObject(value, indent + '  ');
+        } else {
+          result += value + '\n';
+        }
+      }
+    } else {
+      result = indent + obj + '\n';
+    }
+    return result;
+  };
+
+  const formatPublicKey = (publicKey: pki.PublicKey) => {
+    return formatObject(publicKey);
+  };
+
+  const formatSignature = (sig: any) => {
+    const width = 16;
+    const uint8Data = util.binary.raw.decode(sig);
+
+    let wcnt = 0;
+    let result = '';
+    for (let i = 0; i < uint8Data.length; i++) {
+      let hexv = uint8Data[i].toString(16);
+      if (hexv.length === 1) {
+        hexv = '0' + hexv;
+      }
+      result += hexv + ':';
+      wcnt++;
+      if (wcnt === width) {
+        wcnt = 0;
+        result += '\n';
+      }
+    }
+
+    return result;
+  };
+
+  const formatCertInfo = (crt: pki.Certificate, order: number) => {
+    return (
+      <>
+        <Split hasGutter key={'version' + order}>
```

Comment:
`Split` is a layout organizer. Instead it would be better to use components, [`DescriptionList`](
 http://v4-archive.patternfly.org/v4/components/description-list) would be a good candidate.
 
 Why choose components instead of layouts? I think the reason is that PatternFly takes care of the design for us. If we pick components, we ensure that we're having a coherent visual with every RedHat product out there.

While if we're using layouts, we're on our own regarding style, and that would mean in the end to spend more time & energy to fit the look and feel of the portfolio.